libs/asset-viewer/src/lib/components/asset-viewer-files/asset-viewer-files.component.ts

```diff
@@ -0,0 +1,110 @@
+import { HttpClient } from '@angular/common/http';
+import { Component, inject, Input, OnDestroy, OnInit } from '@angular/core';
+import { AppState, fromAppShared, TranslatedValue } from '@asset-sg/client-shared';
+import { isNotNull } from '@asset-sg/core';
+import { AssetFile, AssetFileType, LegalDocItemCode } from '@asset-sg/shared';
+import { AssetId } from '@asset-sg/shared/v2';
+import * as RD from '@devexperts/remote-data-ts';
+import { Store } from '@ngrx/store';
+import { filter, map, Observable, Subscription } from 'rxjs';
+import { displayFileSize, makeTranslatedValueFromItemName } from '../../state/asset-search/asset-search.selector';
+
+@Component({
+  selector: 'ul[asset-sg-asset-viewer-files]',
+  templateUrl: './asset-viewer-files.component.html',
+  styleUrl: './asset-viewer-files.component.scss',
+})
+export class AssetViewerFilesComponent implements OnInit, OnDestroy {
+  @Input({ required: true })
+  assetId!: AssetId;
+
+  @Input({ required: true })
+  files!: AssetFile[];
+
+  @Input({ required: true })
+  type!: AssetFileType;
+
+  private readonly store = inject(Store<AppState>);
+
+  private readonly httpClient = inject(HttpClient);
+
+  public readonly legalDocItems$: Observable<Record<LegalDocItemCode, TranslatedValue>> = this.store
+    .select(fromAppShared.selectRDReferenceData)
```

Comment:
we typically do all of these in a separate selector - then this component does little more than display the finished files including correct translations and locale. 
I imagine there already exists such a selector for the AssetDetail, if you just add it there then the input to this component has everything